### PR TITLE
Replace AppGate with PATH

### PR DESCRIPTION
**Pull request by red-0ne**
This PR introduces the configuration and deployment of the PATH Gateway, removing the AppGate server and updating relevant configurations and scripts.

The most important changes include modifications to environment variables, configuration files, Docker services, and startup scripts.

This PR must be merged in tandem with the [docker-compose-walkthrough PR](https://github.com/pokt-network/poktroll/pull/982)

### Environment and Configuration Changes:
* `.env.sample`: Added new environment variables `PATH_GATEWAY_IMAGE_REPO` and `PATH_GATEWAY_IMAGE_TAG`, and replaced mnemonic keys with private keys for application and gateway.
* `appgate/config/appgate_config.yaml`: Removed the AppGate server configuration.
* `gateway/config/gateway_config.yaml`: Replaced by shannon PATH configuration.

### Docker and Deployment Changes:
* `docker-compose.yml`: Removed the AppGate server service and updated the Gateway service to use the PATH Gateway image and configuration.

### Script Changes:
* `helpers.sh`: Removed the `path_prepare_config` function as it is no longer needed.
* `scripts/start-appgate.sh`: Removed the script for starting the AppGate server.
* `scripts/start-gateway.sh`: Removed the script for starting the Gateway server.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -59,38 +59,19 @@ services:
...
-    depends_on:
-      - full-node
-    volumes:
-      - /etc/ssl/:/etc/ssl/
-      - ./appgate:/home/pocket/.poktroll/
-      - ./scripts/start-appgate.sh:/home/pocket/start-appgate.sh
-
-  # Gateway
+  # PATH Gateway
   gateway:
```
Comment (Olshansk) [inline docker-compose.yml:64]: Should we rename this to `path_gateway` to be consistent with localnet_config?

File: .env.sample [hunk 2]
```diff
@@ -4,20 +4,22 @@ NODE_HOSTNAME=
...
+PATH_GATEWAY_IMAGE_REPO=ghcr.io/buildwithgrove/path
 
-# The version of the image to run off-chain components such as RelayMiner and AppGate.
+# The version of the image to run off-chain components such as RelayMiner and PATH.
 # Adjust the value when need to upgrade off-chain actors.
 POKTROLLD_IMAGE_TAG_OFF_CHAIN=0.0.10 # latest
+PATH_GATEWAY_IMAGE_TAG=main
 POKTROLLD_LOG_LEVEL=info
 
 # Credentials for AppGateServer, Gateway and RelayMiner
```
Comment (Olshansk) [inline .env.sample]: Still seeing `AppGateServer` come up in a few places. 

Also in the `.gitignore`, `network-overview`, prometheus.yml, etc...

Are there any others? Could you double check and update this PR?

File: .env.sample [hunk 3]
```diff
@@ -4,20 +4,22 @@ NODE_HOSTNAME=
 # Deployment env vars
 NETWORK_NAME=testnet-beta
 POKTROLLD_IMAGE_REPO=ghcr.io/pokt-network/poktrolld
+PATH_GATEWAY_IMAGE_REPO=ghcr.io/buildwithgrove/path
 
-# The version of the image to run off-chain components such as RelayMiner and AppGate.
+# The version of the image to run off-chain components such as RelayMiner and PATH.
 # Adjust the value when need to upgrade off-chain actors.
 POKTROLLD_IMAGE_TAG_OFF_CHAIN=0.0.10 # latest
```
Comment (Olshansk) [inline .env.sample]: Update this to 11 plese

File: gateway/config/gateway_config.yaml [hunk 4]
```diff
@@ -1,21 +1,17 @@
...
+    lazy_mode: false
+  gateway_config:
+    gateway_mode: centralized
+    gateway_address: YOUR_PATH_GATEWAY_ADDRESS
+    gateway_private_key_hex: YOUR_PATH_GATEWAY_PRIVATE_KEY
+    owned_apps_private_keys_hex:
+      - YOUR_OWNED_APP_PRIVATE_KEY
+
+services:
+  "F00C":
```
Comment (Olshansk) [inline gateway/config/gateway_config.yaml:17]: Can you add a TODO_TECHDEBT somewhere in this repo to replace `F00C` with something else later?

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: .env.sample (hunk 2 above)
Comment (red-0ne) [reply .env.sample]: Left some occurrences of `AppGate` with a `TODO` until `PATH` metrics are added.

File: docker-compose.yml (hunk 1 above)
Comment (red-0ne) [reply docker-compose.yml:64]: Leaving this as is for now (+TODO) since this needs to be consistent with both `localnet` and grafana dashboards.

**Review by okdas — APPROVED** [review]
This is great. I haven't deployed yet but I'm going to go through the cheat-sheets/walkthroughs soon!

File: grafana/dashboards/network-overview.json [hunk 5]
```diff
@@ -112,7 +112,7 @@
                 {
                     "expr": "sum by (rpc_type, service_id)(rate(appgateserver_relay_requests_total[1m]))",
                     "interval": "",
-                    "legendFormat": "Appgate {{rpc_type}} {{service_id}}",
+                    "legendFormat": "TODO_TECHDEBT: Replace Appgate metrics with PATH once available {{rpc_type}} {{service_id}}",
```
Comment (Olshansk) [inline grafana/dashboards/network-overview.json]:
```suggestion
                    "legendFormat": "TODO_TECHDEBT(@okdas): Replace Appgate metrics with PATH once available {{rpc_type}} {{service_id}}",
```

File: gateway/config/gateway_config.yaml [hunk 6]
```diff
@@ -1,21 +1,18 @@
...
+    lazy_mode: false
+  gateway_config:
+    gateway_mode: centralized
+    gateway_address: YOUR_PATH_GATEWAY_ADDRESS
+    gateway_private_key_hex: YOUR_PATH_GATEWAY_PRIVATE_KEY
+    owned_apps_private_keys_hex:
+      - YOUR_OWNED_APP_PRIVATE_KEY
+
+services:
+  # TODO_TECHDEBT: Replace F00C with the final ethereum serviceID once determined.
```
Comment (Olshansk) [inline gateway/config/gateway_config.yaml]:
```suggestion
  # TODO_TECHDEBT(@Olshansk): Replace F00C with the final ethereum serviceID (or anvil).
```

File: docker-compose.yml [hunk 7]
```diff
@@ -59,38 +58,21 @@ services:
...
-    depends_on:
-      - full-node
-    volumes:
-      - /etc/ssl/:/etc/ssl/
-      - ./appgate:/home/pocket/.poktroll/
-      - ./scripts/start-appgate.sh:/home/pocket/start-appgate.sh
-
-  # Gateway
+  # PATH Gateway
+  # TODO_TECHDEBT: Rename the container to be consistent with both LocalNet and
```
Comment (Olshansk) [inline docker-compose.yml]: If we don't give names, it'll never get done.

If it's actually something we care to have done, give it a DRI (DIrectly Responsible Individual)

```suggestion
  # TODO_TECHDEBT(@okdas): Rename the container to be consistent with both LocalNet and
```

**Review by Olshansk — APPROVED** [review]
I pulled the branch to check for change across the board, but haven't tested yet.

Let's start merging everything in and then iterate on main!

**Pull request merged**